app/scripts/controllers/bridge-status/bridge-status-controller.ts

```diff
@@ -264,14 +276,61 @@ export default class BridgeStatusController extends StaticIntervalPollingControl
     }
   };
 
+  #getSrcTxHash = (bridgeTxMetaId: string): string => {
+    const { bridgeStatusState } = this.state;
+    const srcTxHash =
+      bridgeStatusState.txHistory[bridgeTxMetaId].status.srcChain.txHash;
+
+    if (srcTxHash) {
+      return srcTxHash;
+    }
+
+    // Look up in TransactionController if txMeta has been updated with the srcTxHash, if so, update it here as well
+    const txControllerState = this.messagingSystem.call(
+      'TransactionController:getState',
+    );
+    const txMeta = txControllerState.transactions.find(
+      (tx) => tx.id === bridgeTxMetaId,
+    );
+
+    const { nextState } = this.update((_state) => {
+      _state.bridgeStatusState = {
+        ...bridgeStatusState,
+        txHistory: {
+          ...bridgeStatusState.txHistory,
+          [bridgeTxMetaId]: {
+            ...bridgeStatusState.txHistory[bridgeTxMetaId],
+            status: {
+              ...bridgeStatusState.txHistory[bridgeTxMetaId].status,
+              srcChain: {
+                ...bridgeStatusState.txHistory[bridgeTxMetaId].status.srcChain,
+                txHash: txMeta?.hash,
+              },
+            },
+          },
+        },
+      };
+    });
+
+    const nextSrcTxHash =
+      nextState.bridgeStatusState.txHistory[bridgeTxMetaId].status.srcChain
+        .txHash;
+
+    if (nextSrcTxHash) {
+      return nextSrcTxHash;
+    }
+
+    throw new Error('srcTxHash missing');
+  };
```

Comment:
Do we really need to do a state update in here? Are there downsides to just returning `txMeta?.hash` and letting `#fetchBridgeTxStatus` handle it?

If there's no hash fetchBridgeTxStatus can return early

Reply:
Is it typical for a smart tx to not have a hash right after submission? If it's expected, I also don't think we should throw an error if it's not found. This helps keep our logs/errors actionable

Reply:
Done in [c72ac4e](https://github.com/MetaMask/metamask-extension/pull/28460/commits/c72ac4efe3d3828ff8146a1ff385d110e9ff6800)